src/transform/SourceToAstPass.cpp

```diff
@@ -83,8 +83,24 @@ u1 SourceToAstPass::run( libpass::PassResult& pr )
     Lexer lexer( log, specificationFileStream, std::cout );
     lexer.setFileName( specificationFileName );
 
+    auto specificationName = specificationFileName;
```

Comment:
Please move the code for mapping the file name into a "nice" name into a new function.

Comments would be nice for both cases: 
* why substr between . and / (I assume project files?)
* why substr between start and . (Path without file extension?)